### PR TITLE
[SPARK-2315] Implement drop, dropRight and dropWhile for RDDs, which

### PR DESCRIPTION
**Pull request by erikerlandson**
take RDD as input and return new RDD with elements dropped.

These methods are now implemented as lazy RDD transforms.

### CODE REVIEW

**Comment by AlexNisnevich** [comment]
Have any admins verified this patch? `drop` functionality in RDDs would be very useful to have.

**Comment by JoshRosen** [comment]
@erikerlandson What do you think about releasing this (and maybe #1909) as a library on Maven or http://spark-packages.org?  I'm not sure that this is an API that we necessarily want to put in core yet, but if you publish it as a package then folks would be able to use it with their existing Spark deployments without having to upgrade.  The interface for users could still be pretty nice: just add an implicit class / object or set of implicit conversions, then have users import that.

Spark Packages has a helpful command line tool for creating a project template, which might be a timesaver if you decide to go this route: http://spark-packages.org/package/databricks/spark-package-cmd-tool.

**Comment by erikerlandson** [comment]
Hi @JoshRosen, publishing some of these odds and ends in some form has been on my to-do list for a while.  If there's interest, I can bump it up in priority.

**Comment by AlexNisnevich** [comment]
@JoshRosen @erikerlandson That would be great.

**Comment by erikerlandson** [comment]
@AlexNisnevich 
drop, dropRight and dropWhile are now available on the silex project:
http://silex.freevariable.com/latest/api/#com.redhat.et.silex.rdd.drop.DropRDDFunctions

**Comment by JoshRosen** [comment]
Hey @erikerlandson, since I don't think we're going to merge this functionality into core right now, do you mind closing this issue?  BTW, it would be cool to list Silex on http://spark-packages.org, since that would put the library in front of a lot more users / eyeballs.

**Comment by erikerlandson** [comment]
@JoshRosen Yes, that's fine.   I'll ping @willb about listing silex on spark-packages.org.

File: core/src/main/scala/org/apache/spark/rdd/DropRDDFunctions.scala [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  def drop(n: Int):RDD[T] = {
+    if (n <= 0) return self
+
+    // locate partition that includes the nth element
+    val locate = (partitions: Array[Partition], input: RDD[T], ctx: TaskContext) => {
+      var rem = n
+      var p = 0
+      var np = 0
+      while (rem > 0  &&  p < partitions.length) {
+        np = input.iterator(partitions(p), ctx).length
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/rdd/DropRDDFunctions.scala]: Is it really lazy? I think computation will happen here.

**Pull request closed without merging**